src/rsvc/srv_common.c

```diff
@@ -238,14 +238,22 @@ ds_rsvc_get_attr(struct ds_rsvc *svc, struct rdb_tx *tx, rdb_path_t *path,
 
 		rc = rdb_tx_lookup(tx, path, &key, &iovs[j]);
 
-		if (rc != 0) {
+		if (rc == -DER_NONEXIST) {
+			/* attribute do not exist */
+			iovs[j].iov_buf_len = sizes[i];
+			iovs[j].iov_len = 0;
+			sizes[i] = -1;
```

Comment:
the size is unsigned (size_t). so returning -1 is not really valid here.
can you make it to return 0 for non existent attribute?

Reply:
Well, even if it has been decided that we do not support 0-length attrs I would like to use a non-zero value here, so -1 or 0xff...ff do not differ right ?? If you want to protect against any compiler related issue, may be I can use (size-t)-1 , what do you think ?

Reply:
I'm curious why 0 is not valid to set here if there is no 0-len attrs?
I would say 0xff... is fine if it was; but since it isn't, to me it sound 0 is a cleaner solution from an API / user perspective.
consider the programming aspect.
one needs to check that it's not 0, it's not 0xff (it's between) as to say it's a valid value.

Reply:
I never said 0 is not valid and it is very likely to work, again since we do not support 0-length attrs. I would have preferred using -1/0xff... because it looks more as an error/significant value to me, but in fact I don't really care!!... So let's go for 0 in next commit.

Reply:
i know you didn't say that, my question was why really not making it the value to indicate a non-existent entry.. anyway.. 
im happy to hear other opinions though. i was just stating my take

Reply:
> im happy to hear other opinions though. i was just stating my take

Yes, sure and me too, and this is why I have decided to agree with you because I felt that using 0 was looking much more important for you than for using -1 for me ! ;-)